test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go

```diff
@@ -57,18 +59,19 @@ var _ = BeforeSuite(func() {
 	//updated both sets to ensure there is no overlap
 	latencyIsolatedSet := performancev2.CPUSet("1-9")
 	latencyReservedSet := performancev2.CPUSet("0")
-
+	testlog.Infof("current isolated cpus: %s, desired is %s", string(*initialIsolated), latencyIsolatedSet)
 	totalCpus := cpuset.MustParse(string(latencyIsolatedSet)).Size() + cpuset.MustParse(string(latencyReservedSet)).Size()
 	nodesWithSufficientCpu := nodes.GetByCpuAllocatable(workerNodes, totalCpus)
 	//before applying the changes verify that there are compute nodes with sufficient cpus
-	if len(nodesWithSufficientCpu) != 0 {
-		if *initialIsolated != latencyIsolatedSet || *initialReserved != latencyReservedSet {
-			testlog.Info("Update the isolated and reserved cpus sets of the profile")
-			err = profilesupdate.UpdateIsolatedReservedCpus(latencyIsolatedSet, latencyReservedSet)
-			if err != nil {
-				testlog.Error("could not update the profile with the desired CPUs sets")
-			}
-		}
+	Expect(len(nodesWithSufficientCpu)).NotTo(Equal(0), "found 0 nodes with sufficient cpus %d for the performance profile configuration. Quiting ..", totalCpus)
+
+	profNeedsUpdate := *initialIsolated != latencyIsolatedSet || *initialReserved != latencyReservedSet
+	if profNeedsUpdate {
+		By("Update the isolated and reserved cpus sets of the profile")
+		err = profilesupdate.UpdateIsolatedReservedCpus(latencyIsolatedSet, latencyReservedSet)
+		Expect(err).ToNot(HaveOccurred(), "could not update the profile with the desired CPUs sets")
+	} else {
```

Comment:
You can simplify a bit by removing the else and just log that the configuration is valid.
We know if a change is required by the `By` on line 70